stellar-multitenant/pkg/router/multitenant_data_source_router.go

```diff
@@ -0,0 +1,104 @@
+package router
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/tenant"
+)
+
+var (
+	ErrTenantNotFoundInContext = errors.New("tenant not found in context")
+	ErrNoDataSourcesAvailable  = errors.New("no data sources are available")
+)
+
+type tenantContextKey struct{}
+
+type MultiTenantDataSourceRouter struct {
+	dataSources   sync.Map
+	tenantManager *tenant.Manager
+}
+
+func NewMultiTenantDataSourceRouter(tenantManager *tenant.Manager) *MultiTenantDataSourceRouter {
+	return &MultiTenantDataSourceRouter{
+		tenantManager: tenantManager,
+	}
+}
+
+func (m *MultiTenantDataSourceRouter) GetDataSource(ctx context.Context) (db.DBConnectionPool, error) {
+	tenant, ok := GetTenantFromContext(ctx)
+	if !ok {
+		return nil, ErrTenantNotFoundInContext
+	}
+
+	return m.GetDataSourceForTenant(ctx, *tenant)
+}
+
+// GetDataSourceForTenant returns the database connection pool for the given tenant if it exists, otherwise create a new one.
+func (m *MultiTenantDataSourceRouter) GetDataSourceForTenant(ctx context.Context, tenant tenant.Tenant) (db.DBConnectionPool, error) {
```

Comment:
Aren't we shadowing the `tenant` package when creating a variable with the same name?

Reply:
fixed in c587f7ad1fcff4d57c7e8e2bdc18f4cb18ac700b